### PR TITLE
Handle currency symbols with a period in fromUSDToNumber

### PR DESCRIPTION
**Pull request by Li357**
### Fixed Issues
$ https://github.com/Expensify/Expensify/issues/303703

# Tests
Added a unit test for Bs.S, the Venezuelan currency that has a period in the symbol itself. Before, this was parsing as NaN.

# QA
In OldDot, Verify that when changing Settings > Policies > Reports > Report Currency to Bs.S then going to Settings > Policies > Expenses > Time and setting the Default Hourly Rate to Bs.S6.00 correctly autosaves.

### CODE REVIEW

File: lib/str.js
Comment (Li357) [inline lib/str.js:35]: Before we merge, we should consider if we want to rename this method since it now does double-duty by handling all types of currency strings (symbols + values) instead of just USD.

**Comment by Li357** [comment]
I have read the CLA Document and I hereby sign the CLA

File: __tests__/Str-test.js
Comment (dangrous) [inline __tests__/Str-test.js:101]:
```suggestion
        expect(Str.fromUSDToNumber('Bs.S2.48')).toBe(248);
```
Returns a number not a string, looks like

File: lib/str.js
Comment (dangrous) [inline lib/str.js:31]: given that the line above is longer now I think we can consolidate this to one line.

File: lib/str.js
Comment (dangrous) [inline lib/str.js:38]: Will amounts always have a leading zero? like would we ever pass `$.25` or something like that? If so this might return the wrong value. (I'm not sure on the answer to this)

File: lib/str.js
Comment (dangrous) [reply lib/str.js:35]: Yeah I would. `fromCurrencyToNumber` maybe? Just need to make sure we replace it everywhere

**Review by dangrous — CHANGES_REQUESTED** [review]
Couple comments and questions! GH checks are also a bit sad.

**Comment by Li357** [comment]
Good catches, thanks! Turns out this was a little more annoying than I though. I've changed the regex to match based on the end of the amount string to match either `\d+\.?\d*` for `5` or `5.233` or `5.`, or `\.\d+` for no-leading-zero amounts. I've added more tests for success cases that match the current behavior. There are some cases that now are parsed with this new code, such as `($5.00` returns 500 instead of NaN like before but I don't see that as too important.

**Review by dangrous — APPROVED** [review]
Looks good!

File: lib/str.js
Comment (techievivek) [inline lib/str.js:34]: It looks like changes introduced in this PR have bough this regression https://github.com/Expensify/Expensify/issues/309211

1. If the amount field is not present in the CSV file, it throws an error, originally we used to default to 0.
2. If the amount is negative the amount calculated is `NaN` and further defaulting to 0?

Screenshot 2023-08-21 at 4 41 23 PM
Screenshot 2023-08-21 at 4 42 56 PM
Screenshot 2023-08-21 at 4 58 26 PM
Screenshot 2023-08-21 at 4 58 46 PM

Comment (techievivek) [reply lib/str.js:34]: This almost seems to cause another regression here: https://github.com/Expensify/Expensify/issues/309227

Comment (dangrous) [reply lib/str.js:34]: Can you have a negative per diem? Seems like defaulting to 0 in that case would be correct, right?

For the other, maybe something like changing [this line](https://github.com/Expensify/expensify-common/pull/561/files?diff=unified#diff-eb3aad4f6d64ed33948e40b58a65d583ddf513521d7a3e23871e5f275ac907fcR47) to have a default case of 0?

Will defer to @Li357 though

**Pull request merged**
